Arbie/Contracts/contract.py

```diff
@@ -18,46 +18,61 @@ class Network(Enum):
     ropsten = 2
 
 
+def transact(w3, address: Address, transaction):
+    """Transact a transaction and return transaction receipt."""
+    tx_hash = transaction.transact({
+        'from': address.value,
+        'gas': 48814000,
+    })
+    # wait for the transaction to be mined
+    return w3.eth.waitForTransactionReceipt(tx_hash, 180)  # noqa: WPS432
+
+
 class Contract(object):
     """Base class for contracts."""
 
-    def __init__(self, w3, address: Address, contract):
+    def __init__(self, w3, owner_address: Address, contract):
         self.w3 = w3
-        self.address = address
+        self.owner_address = owner_address
         self.contract = contract
 
+    def get_address(self) -> Address:
+        return Address(self.contract.address)
+
+    def _transact(self, transaction):
+        return transact(self.w3, self.owner_address, transaction)
+
+    def _transact_status(self, transaction) -> bool:
+        return bool(self._transact(transaction).status)
+
 
 class ContractFactory(object):
 
     def __init__(self, w3, factory_class: Contract):
```

Comment:
Contract factory shoul take owner_address in constructor instead of forcing functions to pass it in.

Reply:
Lets change this some other time